apps/worker/tasks/manual_trigger.py

```diff
@@ -96,6 +97,26 @@ def process_impl_within_lock(
         for upload in uploads:
             if not upload.state or upload.state_id == UploadState.UPLOADED.db_id:
                 still_processing += 1
+
+        # Also check Redis processing state to prevent race conditions
```

Comment:
I'm not sure I buy this, I think it makes more sense to use either the DB or Redis as the source of truth even with a race condition.